### DISCUSSION TITLE
Why does giscus need a server (backend)?

### DISCUSSION THREAD

**Discussion by fknfilewalker** [giscus/giscus#688] [Q&A]
I just wonder why giscus needs a server, shouldn't it work just fine with clientside js only?

**Comment by laymonage** [comment]
Because the GraphQL API requires authentication for all requests. Without a server, users need to log in before viewing the discussion. With a server, the API request can be proxied so that it authenticates as the app if the user hasn't signed in.

**Reply by fknfilewalker** [reply]
So only the authentication is done through this server, so no other traffic? I saw with other projects, they first use the free 50 requests per hour and then switch to the server auth for 5000 requests per hour. This does not work here I guess?

Why does the custom css need to be (passed to) loaded by the server when the server is only used for authentication?

**Reply by laymonage** [reply]
There's no "free 50 requests per hour" with the GraphQL API. Sign-in is required. Discussions API is only available through GraphQL.

> Why does the custom css need to be (passed to) loaded by the server when the server is only used for authentication?

This wasn't the case in the initial implementation, but I switched to use SSR for the widget page (the one loaded in the iframe). This was to allow additional server-side validations (e.g. making sure the parent page's domain, i.e. `origin`, is allowed by the repository) instead of just relying on client-side.
